generators/app/templates/src/containers/Profile/profile.container.js

```diff
@@ -0,0 +1,215 @@
+import React, { Component } from "react";
+import { withWebId } from "@inrupt/solid-react-components";
+import { withToastManager } from "react-toast-notifications";
+import data from "@solid/query-ldflex";
+import ProfileShape from "@contexts/profile-shape.json";
+import { ProfileComponent } from "./profile.component";
+
+const defaulProfilePhoto = "/img/icon/empty-profile.svg";
+
+/**
+ * We are using ldflex to fetch profile data from a solid pod.
+ * ldflex libary is using json-LD for this reason you will see async calls
+ * when we want to get a field value, why ? becuase they are expanded the data
+ * this means the result will have a better format to read on Javascript.
+ * for more information please go to: https://github.com/solid/query-ldflex
+ */
+
+export class Profile extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      formFields: [],
+      formMode: true,
+      photo: defaulProfilePhoto
+    };
+  }
+  componentDidMount() {
+    this.fetchPhoto();
+    this.fetchShape();
+  }
+  changeFormMode = () => {
+    this.setState({ formMode: !this.state.formMode });
+  };
+  /**
+   * onChangeInput will update a field into formFields array
+   * and will add updated flag to true, this will be taken
+   * for submit form to check which fields needs to be sent to POD
+  */
+  onInputChange = (e: Event) => {
+    const name = e.target.name;
+    const value = e.target.value;
+
+    const updatedFormField = this.state.formFields.map(field => {
+      if (field.property === name || field.blankNode === name) {
+        return {
+          ...field,
+          updated: true,
+          value
+        };
+      }
+
+      return { ...field };
+    });
+
+    this.setState({ formFields: updatedFormField });
+  };
+  /**
+   * onSubmit will send all the updated fields to POD
+   * fields that was not updated will be not send it.
+  */
+  onSubmit = async (e: Event) => {
+    try {
+      e.preventDefault();
+      let node;
+      const updatedFormField = await Promise.all(
+        this.state.formFields.map(async field => {
+          if (field.updated) {
+            node = data.user[field.property];
+            if (field.blankNode) {
+              node = data[field.nodeParentUri][field.blankNode];
+            }
+            await node.set(field.value);
+
+            return {
+              ...field,
+              updated: false
+            };
+          }
+          return { ...field };
+        })
+      );
+      this.props.toastManager.add("Profile was updated successfully", {
```

Comment:
Instead of setting the toastManager on props, you might want to set it on the React context (https://reactjs.org/docs/context.html). This avoids passing it around everywhere.

Reply:
yep we are using a external library for that, maybe we can create our custom toaster on future

Reply:
My point is rather about not passing it down all the way; but not that important.